pisa/stages/osc/pi_2nu.py

```diff
@@ -0,0 +1,147 @@
+"""
+PISA pi stage for the calculation osc. probabilities assuming two-neutrino model
+
+"""
+from __future__ import absolute_import, print_function, division
+
+import numpy as np
+from numba import guvectorize
+
+from pisa import FTYPE, TARGET
+from pisa.core.pi_stage import PiStage
+from pisa.utils.log import logging
+from pisa.utils.profiler import profile
+
+from pisa.utils.numba_tools import WHERE, myjit
+from pisa.utils.resources import find_resource
+
+
+class pi_2nu(PiStage):
+    """
+    two neutrino osc PISA Pi class
+
+    Paramaters
+    ----------
+    theta23 : quantity (angle)
+    deltam31 : quantity (mass^2)
+
+    None
+
+    Notes
+    -----
+
+    """
+    def __init__(self,
+                 data=None,
+                 params=None,
+                 input_names=None,
+                 output_names=None,
+                 debug_mode=None,
+                 input_specs=None,
+                 calc_specs=None,
+                 output_specs=None,
+                ):
+
+        expected_params = (
+                           'theta23',
+                           'deltam31',
+                          )
+
+        input_names = ()
+        output_names = ()
+
+        # what are the keys used from the inputs during apply
+        input_apply_keys = ('weights',
+                            'nu_flux',
+                            'true_energy',
+                            'true_coszen',
+                           )
+
+        # what keys are added or altered for the outputs during apply
+        output_apply_keys = ('weights',
+                      )
+
+        # init base class
+        super(pi_2nu, self).__init__(
+            data=data,
+            params=params,
+            expected_params=expected_params,
+            input_names=input_names,
+            output_names=output_names,
+            debug_mode=debug_mode,
+            input_specs=input_specs,
+            calc_specs=calc_specs,
+            output_specs=output_specs,
+            input_apply_keys=input_apply_keys,
+            output_apply_keys=output_apply_keys,
+        )
+
+        assert self.input_mode is not None
+        assert self.calc_mode is not None
```

Comment:
it seems like you have no extra calculation, so this can (should) actually be none..?